full-service/src/service/transaction.rs

```diff
@@ -468,6 +468,22 @@ where
         let conn = pooled_conn.deref_mut();
 
         exclusive_transaction(conn, |conn| {
+            if Account::get(&AccountID(account_id_hex.to_string()), conn)?
+                .spend_only_from_subaddress
+            {
+                if subaddress_to_spend_from.is_none() {
+                    return Err(TransactionServiceError::TransactionBuilder(WalletTransactionBuilderError::InvalidArgument(
+                        "This account is configured to spend only from a specific subaddress. Please provide a subaddress to spend from.".to_string()
+                    )));
+                }
+            } else {
+                if subaddress_to_spend_from.is_some() {
+                    return Err(TransactionServiceError::TransactionBuilder(WalletTransactionBuilderError::InvalidArgument(
+                        "This account is not configured to spend only from a specific subaddress. Please do not provide a subaddress to spend from.".to_string()
+                    )));
+                }
```

Comment:
Is this a limitation we actually want to impose?
I could understand as a potential early warning that someone didn't set the spend from subaddress up correctly. 
but inherently it doesn't seem like this is an issue if someone for some reason wanted to drain their change account or similar.

Reply:
Good point. I think that it would be ok to spend from a subaddress even when it is not required.

Reply:
Very good point - I was also thinking about the case of an onramp...

If I create a subaddress for someone, as part of my account establishment, but the first thing they want to do is onramp, then I'm going to need to send eUSD from somewhere...that's not their account. Or even if they had some TXOs, but they're onramping more than they've deposited, I still may presumably want to send them funds from another subaddress (maybe that I set up specifically for this purpose)

Reply:
But actually, both of those cases would be fine even if `require_spend_from_subaddress` was set. Disregard :)

Reply:
Addressed! You can now spend from a subaddress even if the `require spend subaddress` mode was not enabled for that account